### PR TITLE
chore: [WIP] 🍰 New Deployment With 'base' And 'code' Docker Images

### PR DESCRIPTION
**Pull request by Tirokk**
## 🍰 Pullrequest

New deployment with 'base' and 'code' Docker images.

- Push always for test purposes. To be reverted later.
- Define Docker-Compose to build with sender.fm DockerHub images.

### Issues

- relates #1 

### Todo

- [x] set publish again on master push only
- [x] remove comments saying `# TODO build and push "latest-base" and "latest-code“`
- [x] remove line with `# Wolle comment overfluid here?`?
- [x] look after comment `# TODO - this seems not be needed anymore for the new rebranding`

### CODE REVIEW

File: docker/webapp.Dockerfile [hunk 1]
```diff
@@ -1,9 +1,47 @@
...
+
+##################################################################################
+# BRANDED (Does contain only "binary"- and static-files to reduce image size) ####
+##################################################################################
+FROM $APP_IMAGE_BASE as branded
+
+# TODO - do all copying with one COPY command to have one layer
+# Copy "binary"-files from build image
+COPY --from=build ${DOCKER_WORKDIR}/.nuxt ./.nuxt
+COPY --from=build ${DOCKER_WORKDIR}/node_modules ./node_modules
```
Comment (Mogge) [inline docker/webapp.Dockerfile:34]: No

File: docker/webapp.Dockerfile [hunk 2]
```diff
@@ -1,9 +1,47 @@
...
+COPY --from=build ${DOCKER_WORKDIR}/nuxt.config.js ./nuxt.config.js
+# Copy static files
+# TODO - this seems not be needed anymore for the new rebranding
+# TODO - this should be one Folder containign all stuff needed to be copied
+COPY --from=build ${DOCKER_WORKDIR}/config/ ./config/
+COPY --from=build ${DOCKER_WORKDIR}/constants ./constants
+COPY --from=build ${DOCKER_WORKDIR}/static ./static
+COPY --from=build ${DOCKER_WORKDIR}/locales ./locales
+# Copy package.json for script definitions (lock file should not be needed)
+COPY --from=build ${DOCKER_WORKDIR}/package.json ./package.json
```
Comment (Mogge) [inline docker/webapp.Dockerfile:44]: yarn install --prosuction

**Review by Mogge — APPROVED** [review]
I would like to install online the packages required for production

File: docker/backend.Dockerfile [hunk 3]
```diff
@@ -1,10 +1,44 @@
...
+
+##################################################################################
+# BRANDED (Does contain only "binary"- and static-files to reduce image size) ####
+##################################################################################
+FROM $APP_IMAGE_BASE as branded
+
+# TODO - do all copying with one COPY command to have one layer
+# Copy "binary"-files from build image
+COPY --from=build ${DOCKER_WORKDIR}/dist ./dist
+COPY --from=build ${DOCKER_WORKDIR}/node_modules ./node_modules
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply docker/backend.Dockerfile:35]: Is not working see main repo issue https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4471 .

File: docker/backend.Dockerfile [hunk 4]
```diff
@@ -1,10 +1,44 @@
...
+# Copy static files # Wolle comment overfluid here?
+# TODO - externalize the uploads so we can copy the whole folder
+COPY --from=build ${DOCKER_WORKDIR}/public/img/ ./public/img/
+COPY --from=build ${DOCKER_WORKDIR}/public/providers.json ./public/providers.json
+# Copy package.json for script definitions (lock file should not be needed)
+COPY --from=build ${DOCKER_WORKDIR}/package.json ./package.json
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply docker/backend.Dockerfile:41]: Is not working see main repo issue https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4471 .

File: docker/webapp.Dockerfile (hunk 1 above)
Comment (Tirokk) [reply docker/webapp.Dockerfile:35]: Is not working see main repo issue https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4471 .

File: docker/webapp.Dockerfile (hunk 2 above)
Comment (Tirokk) [reply docker/webapp.Dockerfile:45]: Is not working see main repo issue https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4471 .

**Pull request merged**